% Generated by roxygen2 (4.1.1): do not edit by hand
% Please edit documentation in R/JAM.R
\name{JAM}
\alias{JAM}
\title{JAM (Joint Analysis of Marginal statistics)}
\usage{
JAM(marginal.betas = NULL, n = NULL, X.ref = NULL,
  model.space.priors = NULL, beta.priors = NULL,
  beta.prior.partitions = NULL, g.prior = TRUE, tau = NULL,
  enumerate.up.to.dim = 0, full.mcmc.sampling = FALSE, n.iter = 1e+06,
  n.mil.iter = NULL, seed = 1, n.cases = NULL, extra.arguments = NULL,
  save.path = NULL, max.model.dim = -1, use.da.v2 = FALSE,
  trait.variance.ref = NULL)
}
\arguments{
\item{marginal.betas}{Vector of marginal effect estimates to re-analyse with JAM under multivariate models.}

\item{n}{The size of the dataset inwhich the summary statistics were calculated. This must be specified is use.da.v2=TRUE.}

\item{X.ref}{Reference genotype matrix used by JAM to impute the SNP-SNP correlations. If multiple regions are to be
analysed this should be a list containing reference genotype matrices for each region. Individual's genotype
must be coded as a numeric risk allele count 0/1/2. Non-integer values reflecting imputation may be given.
NB: The risk allele coding MUST correspond to that used in marginal.betas. These matrices must each be positive definite and
the column names must correspond to the names of the marginal.betas vector.}

\item{model.space.priors}{Must be specified if model.selection is set to TRUE.
Two options are available. 1) A fixed prior is placed on the proportion of causal
covariates, and all models with the same number of covariates is equally likely. This
is effectively a Poisson prior over the different possible model sizes. A list must
be supplied for `model.space.priors` with an element "Rate", specifying the prior
proportion of causal covariates, and an element "Variables" containing the list of covariates
included in the model search. 2) The prior proportion of causal covariates
is treated as unknown and given a Beta(a, b) hyper-prior, in which case
elements "a" and "b" must be included in the `model.space.priors` list rather
than "Rate". Higher values of "b" relative to "a" will encourage sparsity.
NOTE: It is easy to specify different model space priors for different collections of
covariates by providing a list of lists, each element of which is itself a model.space.prior
list asm described above for a particular subset of the covariates.}

\item{beta.priors}{There are three options:
1) Leave as null, in which case N(0, 1e6) priors are placed on the effects
of confounders and any predictors included in model selection are assigned a
common normal prior with unknown variance.
2) Provide fixed priors for the confounders only.
3) Provide fixed priors for all covariates.
beta.priors (a matrix/data.frame) can be used to provide fixed priors; rows must be named with the corresponding
variable names in the data, and include Guassian prior means and sds in the first and
second columns respectively.}

\item{beta.prior.partitions}{By default, covariates without informative priors are
ascribed a common Gaussian prior with a Unif(0.01,2) hyper-prior on the effect
standard deviation. beta.prior.partitions can be used to partition the covariates
into different prior groups, within each of which exchangeability can be assumed
and a common prior with an independently estimated variance is used.
beta.prior.partitions must be a list with as many elements as desired
partitions. Each element must in turn be a list containing the following named
elements: "Variables" - a list of covariates to be included in this partition,
and "UniformA" and "UniformB" the Uniform hyper parameters. I.e., for a particular
partition, for v in "Variables",
beta[v] | sigma_beta ~ N(0, sigma_beta^2) and
sigma_beta ~ Unif("UniformA", "UniformB").}

\item{g.prior}{Whether to use a g-prior for the beta's, i.e. a multivariate normal
with correlation structure proportional to sigma^2*X'X^-1, which is thought to aid
variable selection in the presence of strong correlation. By default this is enabled.}

\item{tau}{Value to use for sparsity parameter tau (under the tau*sigma^2 parameterisation).
When using the g-prior, a recommended default is max(n, P^2) where n is the number of individuals, and P is the number of predictors.}

\item{enumerate.up.to.dim}{Whether to make posterior inference by exhaustively calculating
the posterior support for every possible model up to this dimension. Leaving at 0 to disable
and use RJMCMC instead. The current maximum allowed value is 5.}

\item{full.mcmc.sampling}{By default JAM only samples models, since the parameters can be analytically integrated out.
If you would like to force JAM to perform full Reversible Jump MCMC of models and parameters (effects and residual), then
set this to TRUE. The posterior summaries can be seen using \code{\link{PrettyResultsTable}}. Note that this option is
forced to false if inference via model enumeration is requested by setting enumerate.up.to.dim>0.}

\item{n.iter}{Number of iterations to run (default is 1e6)}

\item{n.mil.iter}{Number of million iterations to run. Can optionally be used instead of n.iter for convenience, which it will overide if specified.}

\item{seed}{Which random number seed to use in the RJMCMC sampler.}

\item{n.cases}{If the marginal.betas contain log-Odds Ratios, please specify the number of cases
with this option, so that JAM can calculate the case proportion in order to invoke an approximate
transformation between the linear and logistic scales.}

\item{extra.arguments}{A named list of any additional arguments for BGLiMS. Type "data(DefaultArguments)" and look in the
"default.arguments" list to see the names (which must match) and default values of available extra arguments. Currently these
are not documented - please contact the package author, Paul Newcombe for details.}

\item{save.path}{Optional path to save BGLiMS's data and results files. These are usually written as temporary files, and deleted
after running R2BGLiMS. However, this option might help for debugging.}

\item{max.model.dim}{Optional specification of maximum model dimension (default -1 means no maximum is set).}

\item{use.da.v2}{Whether to use Daniel Ahfock's new formulation of the marginal JAM model likelihood. NB: Requires specification of trait.variance.ref and n.}

\item{trait.variance.ref}{Reference estimate of the trait variance. Must specify if use.da.v2=TRUE.}
}
\value{
A Reversible Jump results object is returned. This is a list of two elements: "args" which records various modelling
arguments used in the analysis, and "results" - a matrix containing all saved posterior samples from the analysis. Columns
correspond to parameters and each row contains values from a particular itertation, with 0 indicating exclusion from the model.

The function \code{\link{PrettyResultsTable}} can be used to print summary posterior results for all parameters. Other functions
for summarising results are listed under "see also".
}
\description{
Wrapper for JAM (Joint Analysis of Marginal statistics). See the Vignette.
}
\examples{
library(R2BGLiMS)

### --- Linear regression with full MCMC of all parameters
set.seed(1)
# Simulate data
X <- matrix(rnorm(1000, 0, 1), 100,10) # n=100, 10 covariates
confounder <- rnorm(100, 0, 1) 
b <- c(1, rep(0,9), 4) # Only the first covariate has an effect
y <- rnorm(100, cbind(X,confounder)\%*\%b,1)
data <- data.frame(cbind(X,confounder,y))
# Run model treating prior variance of the effects as unknown
# Poisson model space prior on number of effects
poisson.model.space.prior=list( list("Rate"=0.1, "Variables"=paste("V",c(1:10),sep="")) ) 
lm.hierarchical.prior.results <- R2BGLiMS(
  likelihood="Gaussian", data=data, outcome.var="y", confounders="confounder",
  model.space.priors = poisson.model.space.prior
)
PrettyResultsTable(lm.hierarchical.prior.results)
# A numeric representation of the results is stored in the slot 'posterior.summary.table':
lm.hierarchical.prior.results@posterior.summary.table
# Re-analyse using fixed priors
# (not enough covariates to estimate effect variance)
beta.priors <- data.frame(
  cbind(rep(0,11),rep(10,11)),
  row.names=c("confounder",paste("V",c(1:10),sep="")) )
lm.fixed.prior.results <- R2BGLiMS(
  likelihood="Gaussian", data=data, outcome.var="y", confounders="confounder",
  model.space.priors = poisson.model.space.prior,
  beta.priors=beta.priors)
PrettyResultsTable(lm.fixed.prior.results) # Mixing seems a little poorer
# Re-analyse using the conjugate Gaussian model
# Only models are sampled: effects are analytically integrated out
lm.conjugate.results <- R2BGLiMS(
  likelihood="GaussianConj", data=data, outcome.var="y",confounders="confounder",
  model.space.priors=poisson.model.space.prior,
  enumerate.up.to.dim=0,
  tau=100, # Tau recommended to be maximum of P^2 and N
  g.prior=T)
PrettyResultsTable(lm.conjugate.results)
TopModels(lm.conjugate.results)
# Obtain posterior sample of parameter V1
posterior.sample <- NormInvGamPosteriorSample(
  data=data, outcome.var ="y", confounders="confounder",
  model=c("V1"),tau=100)
# Median and credible interval
round(quantile(posterior.sample[,"V1"],c(0.5, 0.025, 0.975)),2)
# Clearly excludes 0

### --- Logistic regression with two model space prior partitions
# Example dataset from MASS
utils::data(biopsy, package = "MASS")
# Normalise predictors in order to use (default) common unknown priors on their effect sizes
for (v in paste("V",c(1:9),sep="")) {
  biopsy[,v] <- (biopsy[,v]-mean(biopsy[,v],na.rm=T))/sd(biopsy[,v],na.rm=T)
}
biopsyResults <- R2BGLiMS( # Takes a few minutes to run
  likelihood="Logistic", data=biopsy, outcome.var="class",
  model.space.priors=list(
    list("Rate"=0.5, "Variables"=paste("V",c(1:3),sep="")),
    list("Rate"=0.1, "Variables"=paste("V",c(4:9),sep=""))
  )
)
PrettyResultsTable(biopsyResults)
TopModels(biopsyResults)

### --- Survival analysis with priors specified on effect sizes
# Example dataset from MASS
utils::data(VA, package = "MASS")
# Generate priors on log-effect sizes
predictors <- c("treat","age","Karn","diag.time","prior")
beta.priors <- data.frame(
  cbind(rep(0,length(predictors)),rep(10,length(predictors))),
  row.names=predictors)
# Analysis using the Weibull likelihood
# Notice how survival analysis requires additionally flagging a "times.var"
va.results.weibull <- R2BGLiMS(
  likelihood="Weibull",
  data=VA,
  outcome.var="status",
  times.var="stime",
  model.space.priors=list(list("a"=1,"b"=1,"Variables"=predictors)),
  beta.priors=beta.priors
)
PrettyResultsTable(va.results.weibull)
TopModels(va.results.weibull)
# Analysis using the Cox likelihood
va.results.cox <- R2BGLiMS(
  likelihood="Cox",
  data=VA,
  outcome.var="status",
  times.var="stime",
  model.space.priors=list(list("a"=1,"b"=1,"Variables"=predictors)),
  beta.priors=beta.priors
)
PrettyResultsTable(va.results.cox)
TopModels(va.results.cox)
}
\author{
Paul Newcombe
}
\seealso{
Summary results are stored in the slot posterior.summary.table. See also
\code{\link{PrettyResultsTable}} and \code{\link{ManhattanPlot}}. For posterior model space
summaries see \code{\link{TopModels}}. For convergence checks
plots see \code{\link{ChainPlots}} and \code{\link{AutocorrelationPlot}}.
}

